client/admin.go

```diff
@@ -0,0 +1,196 @@
+	Quit = "quit"
+	// User expects proto.UserConfig
+	User = "users"
+	// Zone expects proto.ZoneConfig.
+	Zone = "zones"
+)
+
+// AdminClient issues http requests to admin endpoints.
+// TODO(marc): unify the way we handle addresses in clients.
+type AdminClient struct {
+	context    *base.Context
+	address    string
+	configType string
+}
+
+// NewAdminClient returns a new 'configType' admin client, talking to 'address'.
+func NewAdminClient(ctx *base.Context, address, configType string) *AdminClient {
+	return &AdminClient{ctx, address, configType}
+}
+
+// adminURI build a URI using the passed-in key.
+func (a *AdminClient) adminURI(key string) string {
+	uri := fmt.Sprintf("%s://%s/_admin/%s/%s", a.context.RequestScheme(),
+		a.address, a.configType, url.QueryEscape(key))
+	return uri
+}
+
+// Get issues a GET and returns the plain-text body. It cannot take a key.
+func (a *AdminClient) Get() (string, error) {
+	// Construct our own URI: there is no terminating '/'.
+	uri := fmt.Sprintf("%s://%s/_admin/%s", a.context.RequestScheme(),
+		a.address, a.configType)
+	req, err := http.NewRequest("GET", uri, nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Add(util.AcceptHeader, util.PlaintextContentType)
+	body, err := a.do(req)
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+// GetJSON issues a GET request and returns a json-encoded response.
+func (a *AdminClient) GetJSON(key string) (string, error) {
+	req, err := http.NewRequest("GET", a.adminURI(key), nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Add(util.AcceptHeader, util.JSONContentType)
+	body, err := a.do(req)
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+// GetYAML issues a GET request and returns a yaml-encoded response.
+func (a *AdminClient) GetYAML(key string) (string, error) {
+	req, err := http.NewRequest("GET", a.adminURI(key), nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Add(util.AcceptHeader, util.YAMLContentType)
+	log.Infof("***** Issuing: %+v", req)
+	body, err := a.do(req)
+	log.Infof("Got: %v, %v", body, err)
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+// SetYAML issues a POST request for the given key using the yaml-encoded body.
+func (a *AdminClient) SetYAML(key, body string) error {
+	req, err := http.NewRequest("POST", a.adminURI(key), strings.NewReader(body))
+	if err != nil {
+		return err
+	}
+	req.Header.Add(util.ContentTypeHeader, util.YAMLContentType)
+	_, err = a.do(req)
+	return err
+}
+
+// List issues a GET request to list all configs. Returns a list of keys.
+func (a *AdminClient) List() ([]string, error) {
+	// Construct our own URI: there is no terminating '/'.
+	uri := fmt.Sprintf("%s://%s/_admin/%s", a.context.RequestScheme(),
+		a.address, a.configType)
+	req, err := http.NewRequest("GET", uri, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add(util.AcceptHeader, util.PlaintextContentType)
```

Comment:
Actually, the "return a list on Get /_admin/<type>" doesn't care what the accept header says. That should probably change, but I'm doing client-side only for now. Removing the header.